prereise/gather/griddata/transmission/geometry.py

```diff
@@ -0,0 +1,290 @@
+        self.spacing_L = self.calculate_equivalent_spacing("inductance")
+        self.spacing_C = self.calculate_equivalent_spacing("capacitance")
+
+    def calculate_equivalent_spacing(self, type="inductance"):
+        if type == "inductance":
+            conductor_distance = self.conductor.gmr
+        elif type == "capacitance":
+            conductor_distance = self.conductor.radius
+        else:
+            raise ValueError("type must be either 'inductance' or 'capacitance'")
+        if self.n == 1:
+            return conductor_distance
+        elif self.n == 2:
+            return (conductor_distance * self.spacing) ** (1 / 2)
+        else:
+            if self.layout == "circular":
+                return self.calculate_equivalent_spacing_circular(conductor_distance)
+            if self.layout == "flat":
+                return self.calculate_equivalent_spacing_flat(conductor_distance)
+            raise ValueError(f"Unknown layout: {self.layout}")
+
+    def calculate_equivalent_spacing_circular(self, conductor_distance):
+        if self.n == 3:
+            return (conductor_distance * self.spacing**2) ** (1 / 3)
+        if self.n == 4:
+            return (conductor_distance * self.spacing**3 * 2 ** (1 / 2)) ** (1 / 4)
+        raise NotImplementedError(
+            "Geometry calculations are only implemented for 1 <= n <= 4"
+        )
+
+    def calculate_equivalent_spacing_flat(self, conductor_distance):
+        if self.n == 3:
+            return (conductor_distance * 2 * self.spacing**2) ** (1 / 3)
+        if self.n == 4:
+            return (conductor_distance * 12 * self.spacing**3) ** (1 / 8)
+        raise NotImplementedError(
+            "Geometry calculations are only implemented for 1 <= n <= 4"
+        )
+
+
+@dataclass
+class PhaseLocations(DataclassWithValidation):
+    """Represent the locations of each conductor bundle on a transmission tower. Each of
+    ``a``, ``b``, and ``c`` are the (x, y) location(s) of that phase's conductor(s).
+
+    :param tuple a: the (x, y) location of the single 'A' phase conductor if
+        ``circuits`` == 1, or the ((x1, y1), (x2, y2), ...) locations of the 'A' phase
+        conductors if ``circuits`` > 1. Units are meters.
+    :param tuple b: the (x, y) location of the single 'B' phase conductor if
+        ``circuits`` == 1, or the ((x1, y1), (x2, y2), ...) locations of the 'B' phase
+        conductors if ``circuits`` > 1. Units are meters.
+    :param tuple c: the (x, y) location of the single 'C' phase conductor if
+        ``circuits`` == 1, or the ((x1, y1), (x2, y2), ...) locations of the 'C' phase
+        conductors if ``circuits`` > 1. Units are meters.
+    :param int circuits: the number of circuits on the tower.
+    """
+
+    a: tuple
+    b: tuple
+    c: tuple
+    circuits: int = 1
+    equivalent_distance: float = field(init=False)
+    equivalent_height: float = field(init=False)
+    phase_self_distances: dict = field(init=False)
```

Comment:
Not sure whether we could initialize `phase_self_distances` to be `None` given `circuits` is initialized to be 1, then l.199 can be removed.

Reply:
Yeah! Done.